ext/dom/php_dom.c

```diff
@@ -1494,7 +1494,8 @@ static void dom_libxml_reconcile_ensure_namespaces_are_declared(xmlNodePtr nodep
 	 * Although libxml2 currently does not use this for the reconciliation, it still
 	 * makes sense to do this just in case libxml2's internal change in the future. */
 	xmlDOMWrapCtxt dummy_ctxt = {0};
-	xmlDOMWrapReconcileNamespaces(&dummy_ctxt, nodep, /* options */ 0);
+	bool remove_redundant = nodep->nsDef == NULL && nodep->ns != NULL;
+	xmlDOMWrapReconcileNamespaces(&dummy_ctxt, nodep, /* options */ remove_redundant ? (1 << 0) : 0);
```

Comment:
Should the ``(1 << 0)`` be exposed as a constant somewhere?

Reply:
Sadly libxml2 doesn't have a define for this. I can make my own define (but with a PHP_LIBXML2_ prefix to prevent name collisions). I'll do that in the merge.